app/src/main/java/com/android/unio/model/user/UserRepositoryFirestore.kt

```diff
@@ -18,7 +19,7 @@ class UserRepositoryFirestore @Inject constructor(private val db: FirebaseFirest
 
   override fun init(onSuccess: () -> Unit) {
     Firebase.auth.registerAuthStateListener {
-      if (it.currentUser != null) {
+      if (it.currentUser != null && it.currentUser!!.isEmailVerified) {
```

Comment:
The part that checks if the user is null becomes useless (e.g. `it.currentUser != null`) with this addition. Redundant null check, you should remove this.